src/DataTool.php

```diff
@@ -173,6 +173,29 @@ public function setFields($fieldDefs)
                 continue;
             }
 
+            // enum probability option
+            if (!empty($fieldRules['enum_key_probabilities'])
+                && !empty($GLOBALS['app_list_strings'][$fieldDef['options']])) {
+                $possibleKeys = $GLOBALS['app_list_strings'][$fieldDef['options']];
+                $allProbabilities = $fieldRules['enum_key_probabilities'];
+                $probabilityRest = 100 - array_sum($allProbabilities);
+                $missedProbabilities = array_diff_key($possibleKeys, $allProbabilities);
+                if ($probabilityRest > 0 && !empty($missedProbabilities)) {
+                    $probabilityPerValue = round($probabilityRest / count($missedProbabilities));
+                    foreach ($missedProbabilities as $value) {
+                        $allProbabilities[$value] = $probabilityPerValue;
+                    }
+                }
+
+                // transform probabilities to a [to, $value] fo further usage
+                $summary = 0;
+                foreach ($allProbabilities as $key => $probability) {
+                    $allProbabilities[$key] = [$summary, $key];
+                    $summary += $probability;
+                }
+                $fieldRules['enum_key_probabilities'] = array_reverse(array_values($allProbabilities));
```

Comment:
There are a lot of variables that have `probability` in it's name. Could you please extract the part of the code that calculates enum probabilities to a separate private method `calcEnumProbabilities`. Inside this method you can remove `probabilities` prefix from most of the local variables since it's already clear from the method name that we're working with probabilities.